extensions/data-plane/data-plane-aws-s3/src/test/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSinkTest.java

```diff
@@ -107,6 +112,52 @@ void transferParts_multiPart_succeeds(List inputStream) {
         assertThat(completeMultipartUploadRequest.multipartUpload().parts()).hasSize(2);
     }
 
+    @Test
+    void transferParts_fails_downloading() {
```

Comment:
this test has an integration scope, please remove everything related to S3DataSource and pass to `transferParts` a list with a single mocked `Part` object, that will throw the exception when `openStream` has been called.
Then `S3DataSource.openStream` needs to be tested separately